src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1429 @@
+    dsVideoPortResolution_t resolution;
+    
+    ret = dsHdmiInGetCurrentVideoMode(&resolution);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsHdmiInGetCurrentVideoMode Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    UT_LOG_INFO("Result: Passed dsHdmiInGetCurrentVideoMode resolution name: %s, pixelResolution: %s\t"
+                      "aspectRatio:%s, stereoScopicMode:%s, frameRate:%s, interlaced:%s ",
+                      resolution.name,
+                      ut_control_GetMapString(dsVideoResolution_mapTable, resolution.pixelResolution),
+                      ut_control_GetMapString(dsVideoAspectRatio_mapTable, resolution.aspectRatio),
+                      ut_control_GetMapString(dsVideoStereoScopicMode_mapTable, resolution.stereoScopicMode),
+                      ut_control_GetMapString(dsVideoFrameRate_mapTable, resolution.frameRate),
+                      ut_control_GetMapString(bool_mapTable, resolution.interlaced));
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to check ARC port.
+*
+* This test function is to check selected HdmiInput port on platform is ARC or not.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 008@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_arc_port(void)
+{
+    gTestID = 8;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select;
+    dsHdmiInPort_t iport = dsHDMI_IN_PORT_NONE;
+    bool isarcport = false;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    printf("\n**********Please select port***********\n");
+    readInput(&select);
+    
+    if(select <= 0 || select >numInputPorts)
+    {
+      printf("\n invalid port selected\n");
+    }
+
+    iport = select;
+    ret = dsIsHdmiARCPort(iport , &isarcport);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsIsHdmiARCPort Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    }
+    
+    UT_LOG_INFO("Result : Passed dsIsHdmiARCPort port type: %s, port isarc: %s , portnumber: %d",
+                  ut_control_GetMapString(dsHdmiInPort_mapTable, iport),
+                  ut_control_GetMapString(bool_mapTable, isarcport),iport);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to get edid.
+*
+* This test function is to get edid bytes of  selected HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 009@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_edid(void)
+{
+    gTestID = 9;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int length = 0;
```

Comment:
use int32_t instead of int. address in all places.

Reply:
done